kong/tools/queue.lua

```diff
@@ -287,30 +291,64 @@ function Queue:process_once()
 end
 
 
+local legacy_params_warned = {}
+
+local function maybe_warn(name, message)
+  local key = name .. "/" .. message
+  if ngx.now() - (legacy_params_warned[key] or 0) >= MIN_WARNING_INTERVAL_SECONDS then
+    kong.log.warn(message)
+    legacy_params_warned[key] = ngx.now()
+  end
+end
+
+
 -- This function retrieves the queue parameters from a plugin configuration, converting legacy parameters
--- to their new locations.  The conversion is silently done here, as we're already warning about the legacy
--- parameters being used when validating each plugin's configuration.
+-- to their new locations
 function Queue.get_params(config)
   local queue_config = config.queue or table_new(0, 5)
+
+  if not queue_config.name then
+    queue_config.name = kong.plugin.get_id()
+  end
+
+  if (config.retry_count or null) ~= null and config.retry_count ~= 10 then
+    maybe_warn(
+      queue_config.name,
+      "the retry_count parameter no longer works, please update "
+        .. "your configuration to use initial_retry_delay and max_retry_time instead")
+  end
+
   if (config.queue_size or null) ~= null and config.queue_size ~= 1 then
     queue_config.max_batch_size = config.queue_size
+    maybe_warn(
```

Comment:
Lets add a comment for our future selves _when_ to remove this deprecation warning.